docs/hbase.md

```diff
@@ -0,0 +1,89 @@
+
+## Prerequisites
+
+In order to expose a JMX endpoint, you must set the `com.sun.management.jmxremote.port` system property. It is recommended to also set the `com.sun.management.jmxremote.rmi.port` system property to the same port. To expose JMX endpoint remotely, you must also set the `java.rmi.server.hostname` system property. By default, these properties are set in a Hbase deployment's hbase-env.sh file and the default Hbase installation requires no JMX authentication with JMX exposed locally on 127.0.0.1:8050.
+
+## Configuration
+
+| Field     | Default   | Description |
+| ---     | ---    | ---   |
+| `type`    | required   | Must be `hbase`. |
+| `endpoint`   | `localhost:10101` | The [JMX Service URL](https://docs.oracle.com/javase/8/docs/api/javax/management/remote/JMXServiceURL.html) or host and port used to construct the Service URL. Must be in the form of `service:jmx:<protocol>:<sap>` or `host:port`. Values in `host:port` form will be used to create a Service URL of `service:jmx:rmi:///jndi/rmi://<host>:<port>/jmxrmi`. |
+| `collect_jvm_metrics` | true    | Should the set of support [JVM metrics](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/docs/jvm.md#metrics) also be collected |
+| `username`   | not set by default | The configured username if JMX is configured to require authentication. |
+| `password`   | not set by default | The configured password if JMX is configured to require authentication. |
+| `collection_interval` | `60s`    | A [time.Duration](https://pkg.go.dev/time#ParseDuration) value, such as `30s` or `5m`. |
+
+Example Configuration:
+
+```yaml
+metrics:
+ receivers:
+ hbase:
+  type: hbase
+ service:
+ pipelines:
+  hbase:
+  receivers:
+   - hbase
+```
+
+## Metrics
+In addition to Hbase specific metrics, by default Hbase will also report [JVM metrics](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/docs/jvm.md#metrics)
+
+| Metric                                                 | Data Type  | Unit           | Labels                       | Description |
+| ---                                                    | ---        | ---            | ---                          | ---   |  
+| `hbase.master.region_server.count` | Gauge | `{servers}` | `state` | The number of region servers. |
+| `hbase.master.regions_in_transition.count` | Gauge | `{regions}` |  | The number of regions that are in transition. |
+| `hbase.master.regions_in_transition.over_threshold` | Gauge | `{regions}` |  | The number of regions that have been in transition longer than a threshold time. |
+| `hbase.master.regions_in_transition.oldest_age` | Gauge | `ms` |  | The age of the longest region in transition. |
+| `hbase.region_server.region.count` | Gauge | `{regions}` | `region_server` | The number of regions hosted by the region server. |
```

Comment:
These metrics all show a `region_server` label but the scoping doc says that label is a resource label and so won't be presentt.

Reply:
Due to limitation in the JMX collector, it does not distinguish between normal and resource attributes. So, it will be visible to the customer as a normal attribute. This may change down the round as the collector evolves.

Reply:
We should strip the label using OT config then, if the label always has a single value.